var/spack/repos/builtin/packages/python/package.py

```diff
@@ -348,6 +348,9 @@ class Python(Package):
     # datetime.now(timezone.utc) segfaults
     conflicts("@3.9:", when="%oneapi@2022.2.1:2023")
 
+    # Per discussion https://github.com/spack/spack/pull/44226
+    conflicts("~ctypes", when="@3.12:")
```

Comment:
Maybe add a `msg` stating that ctypes is now required/no longer optional

Reply:
The `ctypes` variant for linux/darwin builds was used to control whether we built the `ctypes` library with an spack managed `libffi` or let python use it's vendored `libffi`. Since recent releases of python no longer vendor `libffi`, we default to `+ctypes` so it might be misleading to say that it's no longer optional since it was not optional before.

Reply:
A cleaner method might be to restrict the `ctypes` variant to `platform==windows` and rework the existing `ctypes` variant to `external-libffi` variant to prevent any confusion.